src/main/java/com/k3project/demo/repository/UserRepository.java

```diff
@@ -1,11 +1,22 @@
 package com.k3project.demo.repository;
 
 import com.k3project.demo.entity.User;
+import com.k3project.demo.service.dto.UserDTO;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
+import java.util.Optional;
 import java.util.UUID;
 
 @Repository
 public interface UserRepository extends JpaRepository<User, UUID> {
+    //TODO pozriet spravny zapis query
```

Comment:
Use it for all methods if it is possible:
Use JPA for querying user by email, firstName, ... dont use custom query(@Query)